rules/internal/utils/extension.go

```diff
@@ -104,6 +104,20 @@ func GetResource(m *desc.MessageDescriptor) *apb.ResourceDescriptor {
 	return nil
 }
 
+// GetResourceFromDescriptor returns the google.api.resource from
+// an arbitrary descriptor. If the descriptor is not a message descriptor,
+// nil is returned.
+func GetResourceFromDescriptor(d desc.Descriptor) *apb.ResourceDescriptor {
```

Comment:
We probably don't need this actually, at least not for this use case.

The parent of a Field will always be a Message, so it is safe (and reasonable to `panic`) if the `desc.Descriptor` from `f.GetParent()` is not a `*desc.MessageDescriptor`.

So I would actually just use  something like `utils.IsResource((f.GetParent()).(*desc.MessageDescriptor))` instead of a `nil` comparison check with this.

Reply:
sure, sounds good.